src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -239,49 +212,150 @@ public void Should_not_crash_when_sink_subscriber_gone()
+            mockTableDataSink.Verify(s => s.AddFactory(trackers[1].Factory, false));
         }
 
-        private ITagger<IErrorTag> CreateTagger(IContentType bufferContentType = null)
+        private IssueTagger CreateTagger(IContentType bufferContentType = null)
         {
-            var mockTextBuffer = new Mock<ITextBuffer>();
-            mockTextBuffer.Setup(b => b.ContentType).Returns(bufferContentType);
-            ITextBuffer textBuffer = mockTextBuffer.Object;
+            var doc = CreateMockedDocument("anyname", bufferContentType);
+            return CreateTaggerForDocument(doc);
+        }
 
+        private IssueTagger CreateTaggerForDocument(ITextDocument document)
+        {
             var mockTextDataModel = new Mock<ITextDataModel>();
+            mockTextDataModel.Setup(x => x.DocumentBuffer).Returns(document.TextBuffer);
             var textDataModel = mockTextDataModel.Object;
 
             var mockTextView = new Mock<ITextView>();
-            mockTextView.Setup(t => t.TextBuffer).Returns(textBuffer);
+            mockTextView.Setup(t => t.TextBuffer).Returns(document.TextBuffer);
             mockTextView.Setup(t => t.TextDataModel).Returns(textDataModel);
+
             ITextView textView = mockTextView.Object;
 
-            return provider.CreateTagger<IErrorTag>(textView, textBuffer);
+            return provider.CreateTagger<IErrorTag>(textView, document.TextBuffer) as IssueTagger;
+        }
+
+        private ITextDocument CreateMockedDocument(string fileName, IContentType bufferContentType = null)
+        {
+            // Text buffer with a properties collection and current snapshot
+            var mockTextBuffer = new Mock<ITextBuffer>();
+            mockTextBuffer.Setup(b => b.ContentType).Returns(bufferContentType);
+            ITextBuffer textBuffer = mockTextBuffer.Object;
+
+            var dummyProperties = new PropertyCollection();
+            mockTextBuffer.Setup(p => p.Properties).Returns(dummyProperties);
+
+            var mockSnapshot = new Mock<ITextSnapshot>();
+            mockSnapshot.Setup(x => x.Length).Returns(0);
+            mockTextBuffer.Setup(x => x.CurrentSnapshot).Returns(mockSnapshot.Object);
+
+            // Create the document and associate the buffer with the it
+            var mockTextDocument = new Mock<ITextDocument>();
+            mockTextDocument.Setup(d => d.FilePath).Returns(fileName);
+            mockTextDocument.Setup(d => d.Encoding).Returns(Encoding.UTF8);
+
+            mockTextDocument.Setup(x => x.TextBuffer).Returns(textBuffer);
+
+            // Register the buffer-to-doc mapping for the factory service
+            dummyDocumentFactoryService.RegisterDocument(mockTextDocument.Object);
+
+            return mockTextDocument.Object;
         }
 
-        private void SetMockDocumentFilename(string filename)
+        private class DummyTextDocumentFactoryService : ITextDocumentFactoryService
```

Comment:
I originally tried to use a _Mock_, but handling the _out_ parameters of the _TryGetDocument_ method was complicated and it was easier just to provide a simple implementation of the interface.